core/vm/evm.go

```diff
@@ -511,7 +511,7 @@ func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gas uint64,
 	// be stored due to not enough gas set an error and let it be handled
 	// by the error checking condition below.
 	if err == nil {
-		if !evm.chainRules.IsEIP4762 {
+		if !evm.chainRules.IsEIP4762 || replayMode {
 			createDataGas := uint64(len(ret)) * params.CreateDataGas
 			if !contract.UseGas(createDataGas) {
```

Comment:
Here's the "bug" for replaying old data (and we need this for the new data too).

The problem is that Eip4762 is active in Prague (thus active in the replay). We don't get gas impact changes since we do the usual `ClearVerkleGasCosts()`, **but** EIP-4762 also removes costs that shouldn't be removed for replay.

In particular, the CALL cost are removed but we need them for replay.

In order to save the emv.chainRules.IsEIP4762 flag, I did a very ugly hack of defining a `replaymode = true` and **only** in the places where `IsEIP4762` is used for **non witness** charging (i.e: removing gas costs), I use this `replayMode` to disable the effect of the flag.

Maybe not the best solution -- just pushing enough changes to know we're able to replay stuff. We can try thinking a good solution for being able merge the replay branch eventually.

Reply:
I actually wonder if wouldn't be better to simply disable `IsEIP4762` in replay mode. Would we losing any important thing being tested with that?

I'm pretty sure the other tx that now fails in this same block is related with a probably related similar bug who knows where. I can try tomorrow though.

Reply:
> I actually wonder if wouldn't be better to simply disable IsEIP4762 in replay mode. Would we losing any important thing being tested with that?

unfortunately, yes: a lot of stuff would not end up in the `AccessWitness` and so the performance would look better than it really is